lib/rspec/core/configuration.rb

```diff
@@ -781,23 +783,62 @@ def full_backtrace=(true_or_false)
       # Returns the configuration option for color, but should not
       # be used to check if color is supported.
       #
+      # @deprecated No longer recommended because of complex behavior. Instead,
+      #   rely on the fact that TTYs will display color by default, or use
+      #   {#force_color} to display color on a non-TTY output.
       # @see color_enabled?
       # @return [Boolean]
       def color
         value_for(:color) { @color }
       end
 
+      # Returns the configuration option for forcing color, but should not
+      # be used to check if color is supported.
```

Comment:
It would be nice if the docs for this (and the other methods like `color` and `no_color`) described what they do.  "Returns the configuration option for forcing color" is pretty obvious already, and I imagine a user checking the docs for this would want to understand why they might ever want to use this config option.  How about we change these to something like:

``` ruby
# Enables color output if the output is a TTY.  As of RSpec 3.6, this is the default
# behavior and this option is retained only for backwards compatibility.
#
# @depreacted #...
#
# @see force_color
# @see no_color
# @see color_enabled?
# @return [Boolean]
def color
  # ...
end

# Forces the output to be in color, even if the output is not a TTY.
#
# @note By default, RSpec will produce color if the output is a TTY, so this option
#   is really only needed if your output is a non-TTY (or sometimes a non-TTY).
# @see color
# @see no_color
# @see color_enabled?
# @return [Boolean]
def force_color
  # ...
end

# Prevents RSpec from producing color output, regardless of whether or not the
# output is a TTY.
#
# @see force_color
# @see color
# @see color_enabled?
# @return [Boolean]
def no_color
  # ...
end
```